### PR TITLE
chore: Generalize results on `finrank` to rings.

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/FreeModule/Finite/Basic.lean [hunk 1]
```diff
@@ -31,6 +31,13 @@ section Ring
 
 variable [Ring R] [AddCommGroup M] [Module R M] [Module.Free R M]
 
+/-- If a free module is finite, then any arbitrary basis is finite. -/
+lemma finite_basis {R M} [Ring R] [Nontrivial R] [AddCommGroup M] [Module R M]
+    [Module.Free R M] {ι} [Module.Finite R M] (b : Basis ι R M) :
+    _root_.Finite ι := by
+  obtain ⟨s, hs⟩ := ‹Module.Finite R M›
+  exact basis_finite_of_finite_spans (↑s) s.finite_toSet hs b
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/FreeModule/Finite/Basic.lean:39]: nit: I think you can also write this
```suggestion
    _root_.Finite ι :=
  let ⟨s, hs⟩ := ‹Module.Finite R M›
  basis_finite_of_finite_spans (↑s) s.finite_toSet hs b
```

File: Mathlib/LinearAlgebra/FiniteDimensional.lean [hunk 2]
```diff
@@ -823,10 +862,11 @@ end LinearEquiv
 section
 
 variable [DivisionRing K] [AddCommGroup V] [Module K V]
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/FiniteDimensional.lean:864]:
```suggestion
```
nothing in this section uses these any more

File: Mathlib/LinearAlgebra/FiniteDimensional.lean [hunk 3]
```diff
@@ -734,19 +773,19 @@ finite-dimensional.
...
       _ fun i _ => by infer_instance
   · intro S₁ hS₁ S₂ hS₂
     exact Submodule.finiteDimensional_sup S₁ S₂
 #align submodule.finite_dimensional_finset_sup Submodule.finiteDimensional_finset_sup
 
 /-- The submodule generated by a supremum of finite dimensional submodules, indexed by a finite
 sort is finite-dimensional. -/
-instance finiteDimensional_iSup {ι : Sort*} [Finite ι] (S : ι → Submodule K V)
-    [∀ i, FiniteDimensional K (S i)] : FiniteDimensional K ↑(⨆ i, S i) := by
+instance finiteDimensional_iSup {ι : Sort*} [Finite ι] (S : ι → Submodule R V)
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/FiniteDimensional.lean:787]:
```suggestion
instance Module.Finite.iSup {ι : Sort*} [Finite ι] (S : ι → Submodule R V)
```

File: Mathlib/LinearAlgebra/Dimension.lean [hunk 4]
```diff
@@ -1226,7 +1261,7 @@ section DivisionRing
 
 variable [DivisionRing K] [AddCommGroup V] [Module K V] [AddCommGroup V₁] [Module K V₁]
 
-variable [AddCommGroup V'] [Module K V']
+variable [AddCommGroup V'] [Module K V'] [Ring R] [Module R V]
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Dimension.lean:1264]: Surely this change has no effect, as no statements below reference `R`?

**Review by riccardobrasca — COMMENTED** [review]
I think that a sensible thing to do is to move all the results about a general ring (maybe with some rank condition) to a folder inside `FreeModule`, and keep here only results about division rings, but this is of course for another PR. Maybe add a TODO?

**Comment by erdOne** [comment]
Closed in favor of #8912

**Pull request closed without merging**